ext/bson/native-endian.h

```diff
@@ -26,6 +26,16 @@
 #define BSON_BIG_ENDIAN    4321
 #define BSON_LITTLE_ENDIAN 1234
 
+/* See a similar check in ruby's sha2.h */
+# ifndef BSON_BYTE_ORDER
```

Comment:
It doesn't look like BSON_BYTE_ORDER will ever be defined here. It looks like this block was in the right place before.